webui/react/src/services/stream/stream.ts

```diff
@@ -0,0 +1,151 @@
+import { forEach, reduce, trimEnd } from 'lodash';
+
+import { decode_keys, KeyCache } from './keyCache';
+import { StreamSpec } from './projects';
+
+import { Streamable, StreamEntityMap } from '.';
+
+// About 60 seconds of auto-retry.
+const backoffs = [0, 1, 2, 4, 8, 10, 10, 10, 15];
+const sleep = (s: number) => new Promise((r) => setTimeout(r, 1000 * s));
```

Comment:
why not use the `function sleep(s: number): Promise { ... }` syntax here?

I don't know our typescript code... is there a standard on which one to use?